vector/array.c

```diff
@@ -0,0 +1,134 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "array.h"
+
+array *vector_create() {
+    array *arr = vector_create_size(init_capacity);
+
+    if (!arr)
+        return NULL;
+
+    arr->capacity = init_capacity;
+    arr->size = 0;
+
+    return arr;
+}
+
+bool vector_append(array *arr, data_t element) {
+    if (!arr)
+        return false;
+
+    else if (arr->size == arr->capacity) {
+        arr->capacity *= 2;
+        data_t *bigger_array = realloc(arr->array, arr->capacity * sizeof(data_t));
+
+        if (bigger_array != NULL)
+            arr->array = bigger_array;
+        else
+            return false;
+    }
+
+    arr->array[arr->size] = element;
+    arr->size += 1;
+    return true;
+}
+
+void vector_print(array *arr) {
+
+    if (!arr)
+        printf("The array is empty!\n");
+    else {
+        printf("[%d", arr->array[0]);
+        for (size_t i = 1; i < arr->size; ++i)
+            printf(", %d", arr->array[i]);
+        printf("] (size: %lu, capacity: %lu)\n", arr->size, arr->capacity);
+    }
+}
+
+void vector_free(array *arr) {
+    free(arr->array);
+    free(arr);
+}
+
+bool vector_pop(array *arr, data_t* popped_el) {
+    if (!arr)
+        return false;
+    if (arr->size == 0)
+        return false;
+    size_t index = arr->size - 1;
+    *popped_el = arr->array[index];
+    arr->array[index] = 0;
```

Comment:
This is redundant.

(There are use cases for it, like when you would wipe out secret password from the memory or sth like that, but we don't need that in the `vector` library itself).